language-support/hs/bindings/src/DA/Ledger/Jwt.hs

```diff
@@ -0,0 +1,23 @@
+-- Copyright (c) 2019 The DAML Authors. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module DA.Ledger.Jwt(
+  Jwt,
+  tryCreateFromString,
+  toString
+  ) where
+
+import Data.List.Extra (splitOn)
+
+newtype Jwt = Jwt { toString :: String }
+
+tryCreateFromString :: String -> Either String Jwt
+tryCreateFromString s = do
+  validate3parts s
+  return $ Jwt s
+
+validate3parts :: String -> Either String ()
```

Comment:
Is the only purpose of this validation logic to catch easy user errors before sending the token to the ledger and therefore potentially provide better error messages? If so, would be good to point this out in a comment.